packages/host/app/resources/adoption-chain-manager.ts

```diff
@@ -0,0 +1,103 @@
+import { ImportResource } from './import';
+import { Resource } from 'ember-resources';
+import { tracked } from '@glimmer/tracking';
+import { Loader } from '@cardstack/runtime-common/loader';
+import { getOwner } from '@ember/application';
+import { type BaseDef } from 'https://cardstack.com/base/card-api';
+import type LoaderService from '../services/loader-service';
+import {
+  getCardType,
+  type CardType,
+} from '@cardstack/host/resources/card-type';
+import { restartableTask } from 'ember-concurrency';
+import { action } from '@ember/object';
+
+interface AdoptionChainManagerArgs {
+  named: { importResource: ImportResource | undefined; loader: Loader };
+}
+
+export type ElementInFile = CardInFile; // can add more types here
+
+interface CardInFile {
+  cardType: CardType;
+  card: typeof BaseDef;
+}
+
+// forwards the card type resource (doesn't care about the card type resource)
+export class AdoptionChainManager extends Resource<AdoptionChainManagerArgs> {
```

Comment:
I'm just trying to understand what this resource is actually doing--`Manager` is such a overloaded term. it's doesn't really help me to know what this thing does. Reading thru this i think what this resource is doing is trying to load all the cards specifically from a module? Maybe `CardsFromModuleResource` would be a better name for this? the `isSelected` is super confusing--that feels like it should be part of a component, not this resource.

Reply:
Hmmmm i will say that this resource is in charge of managing the selection of elements (elements being exported cards, unexported cards, exported function, exported class). What I intended for this resource is to be consumed by both the detailPanel & the schemeEditor. Perhaps, at one point it will be consumed by monaco to get the location of the selection. 

Would it be better to kill the resource and put things in code mode? Or Rename it?

Reply:
that's super confusing to me--why do you need a resource to manage selection? that seems to be something that fits more naturally in a component not a resource. The component should use the resource to understand what all cards exist, and the selection state should live in the component since that is the class that contains the template that consumes the action for selection

Reply:
Actually the existence of cards wud be handled by the module resource. And i think it will continue that way. Ok in that case, i will remove this resource and put all logic in the root code mode component

Reply:
Ive moved the select logic in code. And I also moved the array of cardTypes into code mode but still in a resource pattern. I think this is what we want tho. If it wasn't a resource, we would be recreating cardType everytime. So like you said we still need to load the cards. But I do agree that having it as adoption-chain-manager was confusing because it was not clear what it was doing